### PR TITLE
Refactor: Replace deferred git commands with regular git commands

### PR DESCRIPTION
**Pull request by sandr01d**
## Check list

- [X] I have performed a self-review of my code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Description

We were using deferred code in git commands in some places without any apparent reason. Each of these deferred code snippets was only executed a single time, so we can replace them with regular git commands.
This PR replaces the remaining git commands that were not removed in #343, because they were not moved into functions.

## Note

This PR belongs to #326 and resulted from discussions in #324.

## Type of change

- [ ] Bug fix
- [ ] New feature
- [X] Refactor
- [ ] Breaking change
- [ ] Documentation change

## Test environment

- Shell
    - [ ] bash
    - [x] zsh
    - [ ] fish
- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Others:

### CODE REVIEW

**Comment by sandr01d** [comment]
This should be the last big PR. After this, only small changes are left :rocket:

File: bin/git-forgit [hunk 1]
```diff
@@ -180,52 +181,41 @@ _forgit_diff_view() {
...
-    _forgit_diff_git_opts=()
-    _forgit_parse_array _forgit_diff_git_opts "$FORGIT_DIFF_GIT_OPTS"
-    git_diff="git diff --color=always ${_forgit_diff_git_opts[*]} $escaped_commits"
-    enter_cmd="cd '$repo' && $get_files | xargs -0 $git_diff -U$_forgit_fullscreen_context -- | $_forgit_diff_pager"
+    for commit in "${commits[@]}"; do
+        escaped_commits+="'${commit//\{/\\\\\{}' "
+    done
     opts="
         $FORGIT_FZF_DEFAULT_OPTS
-        +m -0 --bind=\"enter:execute($enter_cmd | $_forgit_enter_pager)\"
```
Comment (carlfriedrich) [inline bin/git-forgit:154]: I think the `enter_cmd` change here and everything related to it does not fit with the commit description. I would suggest removing it from this commit and do it later together with the same thing in `_forgit_log`, where the `enter_cmd` still exists.

Comment (sandr01d) [reply bin/git-forgit:154]: Sure, done now.

File: bin/git-forgit [hunk 2]
```diff
@@ -716,9 +719,12 @@ _forgit_branch_delete() {
...
 
-    cmd="git branch --color=always | LC_ALL=C sort -k1.1,1.1 -rs"
-    for branch in $(eval "$cmd" | FZF_DEFAULT_OPTS="$opts" fzf | awk '{print $1}'); do
-        _forgit_git_branch_delete "$branch"
+    for branch in $(git branch --color=always |
+        LC_ALL=C sort -k1.1,1.1 -rs |
+        FZF_DEFAULT_OPTS="$opts" fzf |
+        awk '{print $1}')
+    do
+        _forgit_branch_delete "$branch"
```
Comment (carlfriedrich) [inline bin/git-forgit:727]: I think this should stay `_forgit_git_branch_delete`, shouldn't it?

**Review by carlfriedrich — CHANGES_REQUESTED** [review]
@sandr01d Overall this looks great for me, thanks for the work again. Only thing I would like to see is a note in the commit message about why the `FORGIT_LOG_GRAPH_ENABLE` to `_forgit_log_graph_enable` change has been necessary along with this. (I assume it is necessary for this commit? Otherwise I would prefer to have a dedicated commit for it.)

File: bin/git-forgit (hunk 2 above)
Comment (sandr01d) [reply bin/git-forgit]: Yes, good catch!

**Review by carlfriedrich — APPROVED** [review]

**Comment by carlfriedrich** [comment]
@sandr01d Perfect, thanks for the update! Will you do the `enter_cmd` changes in a next PR?

**Comment by sandr01d** [comment]
Changes have been merged with #326

**Pull request closed without merging**